### PR TITLE
RFD 00: RFDs

### PR DESCRIPTION
**Pull request by awly**
Belated doc describing the RFD format and process.
As always, everything is up for discussion.

Also, updated prior RFDs to conform to the style.

### CODE REVIEW

File: rfd/0018-rfds.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+https://oxide.computer/blog/rfd-1-requests-for-discussion/ which is in turn
+inspired by https://www.ietf.org/standards/rfcs/
+
+## Details
+
+Each RFD is stored in a markdown file under
+https://github.com/gravitational/teleport/tree/master/rfd and has a unique
+number.
+
+### structure
```
Comment (webvictim) [inline rfd/0018-rfds.md:39]:
```suggestion
### Structure
```

File: rfd/0018-rfds.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+1. a header containing author name(s) and [state](#states)
+1. title in the format `RFD $NUMBER - $TITLE`
+1. the `What` section - 1-3 sentence summary of what this RFD is about
+1. the `Why` section - a few paragraphs describing motivation for the RFD
+1. the `Details` section - detailed description of the proposal, including
+   APIs, UX examples, migrations or any other relevant information
+
+Use this RFD as an example.
+
+### process
```
Comment (webvictim) [inline rfd/0018-rfds.md:52]:
```suggestion
### Process
```

File: rfd/0018-rfds.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+1. merge the PR and start implementing
+
+1. once implemented, make another PR changing the `state` to `implemented` and
+   updating any details that changed during implementation
+
+If an RFD is eventually deprecated (e.g. a feature is removed), make a PR
+changing the `state` to `deprecated` and optionally link to the replacement RFD
+(if applicable).
+
+#### states
```
Comment (webvictim) [inline rfd/0018-rfds.md:92]:
```suggestion
#### States
```

**Review by webvictim — APPROVED** [review]

**Comment by fspmarshall** [comment]
I feel very strongly that this should be RFD `0000` instead of RFD `0018` :smile:

**Review by benarent — APPROVED** [review]
👍 but I do agree with @fspmarshall meta comment about naming this 00

File: rfd/0018-rfds.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
+---
+authors: Andrew Lytvynov (andrew@gravitational.com)
```
Comment (webvictim) [inline rfd/0018-rfds.md:2]:
```suggestion
authors: Andrew Lytvynov (andrew@goteleport.com)
```

**Pull request merged**